### ISSUE TITLE
Consider making ItemDetails.type extensible

### ISSUE THREAD

**Issue by benjycui** [WICG/digital-goods#49]
ItemDetails.type only supports two values, e.g. "product" and "subscription".

```ts
enum ItemType {
  "product",
  "subscription",
};
```

What if a store provides more types of digital products? In Microsoft Store, there are [five types of products](https://learn.microsoft.com/en-us/uwp/api/windows.services.store.storeproduct.productkind?view=winrt-22621#remarks).

So, do you think that it is possible to make ItemDetails.type extensible?

**Comment by rsolomakhin** [comment]
Yes, that sounds reasonable. Please go ahead and implement the `ItemType` enums that you need. We can add them to the spec at that point.
